### PR TITLE
Add support for json messages and no compression

### PR DESCRIPTION
**Pull request by manugarri**
This PR implements the following changes:

- Supports json encoded messages
- Supports uncompressed messages
- P3 compatible
- Minor fixes (remove commented code and unused imports)

I set all the default encoding to `msgpack` and the decompress to `True`. That should make it compatible with current use cases.

### CODE REVIEW

File: kafka_scanner/msg_processor_handlers.py [hunk 1]
```diff
@@ -20,11 +20,23 @@ def retry_on_exception(exception):
 
 
 class MsgProcessorHandlers(object):
-    def __init__(self, encoding=None):
-        self.decompress_fun = zlib.decompress
+    def __init__(self, encoding=None, decompress=None, msgformat=None):
+        if decompress:
+            import zlib
```
Comment (kalessin) [inline kafka_scanner/msg_processor_handlers.py]: why lazy import of zlib? It is a standard library.

File: kafka_scanner/tests/__init__.py [hunk 2]
```diff
@@ -16,16 +17,25 @@
 ConsumerRecord = namedtuple("ConsumerRecord", ["partition", "offset", "key", "value"])
 
 
-def get_kafka_msg_samples(msgs=None, fetch_count=0):
+def get_kafka_msg_samples(msgs=None, fetch_count=0, msgformat='msgpack', compress=True):
     if not msgs:
-        msgs = [('AD12345', "my-body"),
-                ('AD34567', "second my-body"),
```
Comment (kalessin) [inline kafka_scanner/tests/__init__.py]: Can you separate pylint style fixes and new feature in different commits?

File: kafka_scanner/msg_processor.py [hunk 3]
```diff
@@ -2,9 +2,9 @@
 
 class MsgProcessor(object):
 
-    def __init__(self, handlers_list, encoding=None):
+    def __init__(self, handlers_list, encoding=None, *args, **kwargs):
         self._handlers = []
-        self.processor_handlers = MsgProcessorHandlers(encoding=encoding)
+        self.processor_handlers = MsgProcessorHandlers(encoding=encoding, *args, **kwargs)
```
Comment (kalessin) [inline kafka_scanner/msg_processor.py]: it is weird that positional arguments *args come after a key argument encoding=. Didn't pylint complain here?

File: kafka_scanner/msg_processor_handlers.py [hunk 4]
```diff
@@ -20,11 +20,23 @@ def retry_on_exception(exception):
...
+            self.decompress_fun = zlib.decompress
+        else:
+            self.decompress_fun = self.dummy_decompress
         self.consumer = None
         self.__next_messages = 0
         self.__encoding = encoding
+        self.deserialize_fun = self.set_deserializer(msgformat)
+
+    def set_deserializer(self, msg_format):
```
Comment (kalessin) [inline kafka_scanner/msg_processor_handlers.py]: in the way you are using this method, seems better a name like get_deserializer(). The method doesn't really set anything. Only returns an object than it is set in previous line.

File: tests/test_kafka_scanner.py [hunk 5]
```diff
@@ -378,24 +377,46 @@ def test_kafka_scan_resume_simple_partition_after_fail(self, kafka_consumer_mock
...
+        self.assertEqual(len(set(msgsdict)), 900)
+        self.assertEqual(scanner.scanned_count, 1100)
+        self.assertEqual(scanner.issued_count, 900)
+        self.assertEqual(scanner.deleted_count, 100)
+        self.assertEqual(scanner.dupes_count, 100)
+        self.assertEqual(scanner.test_count, 900)
+        for i in range(100, 200):
+            self.assertTrue('AD%.3d' % i not in msgsdict)
 
-#     def test_kafka_scan_resume(self, kafka_consumer_mock, batchsize=100):
```
Comment (kalessin) [inline tests/test_kafka_scanner.py]: please, don't remove this test.

**Review by kalessin — CHANGES_REQUESTED** [review]

File: kafka_scanner/msg_processor_handlers.py (hunk 1 above)
Comment (manugarri) [reply kafka_scanner/msg_processor_handlers.py]: sure, i can import it, there is no point in polluting the global namespace unnecesarily but it wouldnt be a big deal.

File: kafka_scanner/msg_processor_handlers.py (hunk 4 above)
Comment (manugarri) [reply kafka_scanner/msg_processor_handlers.py]: good point, naming is important

File: kafka_scanner/tests/__init__.py (hunk 2 above)
Comment (manugarri) [reply kafka_scanner/tests/__init__.py]: oh ok, do you guys dont mind if there are multiple commits per PR? or should i remove it completely and then do another PR?

Comment (kalessin) [reply kafka_scanner/tests/__init__.py]: I don't mind multiple commits per PR. Not needed an extra PR.

File: tests/test_kafka_scanner.py (hunk 5 above)
Comment (manugarri) [reply tests/test_kafka_scanner.py]: Actually, now that I see, [this whole code block is commented in master](https://github.com/scrapinghub/kafka-scanner/blob/master/tests/test_kafka_scanner.py#L382). Are you sure you want to keep it in master? Saving code that you might use in the future is not  good practice (YAGNI probably), and that is what git history is for.

File: kafka_scanner/msg_processor.py (hunk 3 above)
Comment (manugarri) [reply kafka_scanner/msg_processor.py]: no, afaik this is perfectly valid. If it goes against SH style guide let me know and I can make it an unnamed argument (i always prefer using names for the shake of documenting but I see your point).

**Comment by manugarri** [comment]
@kalessin Fixed the changes, and commented on your requests. Let me know if you need anything else. We are using kafka scanners and I'd rather not keep our own fork when this change could  benefit everyone instead.

File: tests/test_kafka_scanner.py (hunk 5 above)
Comment (kalessin) [reply tests/test_kafka_scanner.py]: you are right. I hand't noticed this was commented test.

**Comment by kalessin** [comment]
thanks @manugarri . I merged and updated/tagged version. New version just published into pypi.

**Comment by kalessin** [comment]
https://pypi.python.org/pypi/kafka-scanner/0.3.4

**Comment by manugarri** [comment]
yaaay! Thanks @kalessin

**Pull request merged**